pkg/pull/pull.go

```diff
@@ -437,12 +437,15 @@ func Pull(upstreamURI string, pullOptions PullOptions) (string, error) {
 	// for writing Common Midstream, every chart and subchart is in this map as Helm Midstreams will be processed later in the code
 	commonWriteMidstreamOptions.UseHelmInstall = map[string]bool{}
 	for _, v := range newHelmCharts {
-		chartBaseName := v.Spec.Chart.Name
+		chartBaseName := v.GetDirName()
 		// the helmBase may have a chart name prefix removed - we must find the base name instead of the original chart name
 		for _, helmBase := range helmBases {
-			chartName := strings.Split(helmBase.Path, "/")[len(strings.Split(helmBase.Path, "/"))-1]
-			if strings.HasSuffix(v.Spec.Chart.Name, chartName) {
+			baseDir := filepath.Dir(helmBase.Path)
+			chartName := filepath.Base(helmBase.Path)
+			// v is a top level chart, so only top level charts directory name should match
+			if baseDir == "charts" && strings.HasSuffix(chartBaseName, chartName) {
```

Comment:
why not
```suggestion
			if baseDir == "charts" && chartBaseName == chartName {
```

Reply:
This makes this loop pointless.  I don't actually know what it does.  How about just removing it?

Reply:
yes, i agree. we should remove it

Reply:
it does seem pointless indeed. no objection on removing it